frontend/src/components/MapLegend.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+interface LegendProps {
+  modeColors: { [key: string]: string };
+}
+
+const MapLegend: React.FC<LegendProps> = ({ modeColors }) => {
+  return (
+    <div className="flex flex-col mt-3">
+      <div className="font-bold text-md mb-2">Legend</div>
+      <div className="flex flex-wrap justify-center">
+        {Object.entries(modeColors).map(([mode, color]) => (
+          <div key={mode} className="flex items-center mb-2 mr-4">
```

Comment:
```suggestion
          <div key={mode} className="flex items-center mb-2 mx-2">
```
Sorry, idk how I didn't notice this before. but add the margin around each item in the map bc it looks right-aligned when its only applying to the right. this should be the last change